tools/validators/instance_validator/instance_parser.py

```diff
@@ -0,0 +1,113 @@
+                               'type': syaml.Str(),
+                               'id': syaml.Str(),
+                               syaml.Optional('connections'):
+                               syaml.MapPattern(syaml.Str(),
+                                                syaml.Str()) |
+                               syaml.Seq(
+                                   syaml.MapPattern(syaml.Str(),
+                                                    syaml.Str())),
+                               syaml.Optional('links'): syaml.MapPattern(
+                                   syaml.Str(),
+                                   syaml.MapPattern(syaml.Str(), syaml.Str())),
+                               syaml.Optional('translation'): syaml.Any(),
+                               syaml.Optional('metadata'): syaml.Any()
+                               }))
+
+"""schema separately parses translation to account for multiple valid formats
+github.com/google/digitalbuildings/blob/master/ontology/docs/building_config.md#defining-translations"""
+_TRANSLATION_SCHEMA = syaml.Str() | syaml.Any()
+
+# TODO add manual check to de-duplicate units/unit_values/states
+# TODO add all units/unit_values/states to translation_data_schema
+"""further account for multiple valid translation formats
+github.com/google/digitalbuildings/blob/master/ontology/docs/building_config.md#defining-translations"""
+_TRANSLATION_DATA_SCHEMA = syaml.Str() | syaml.Map({
+    'present_value': syaml.Str(),
+    syaml.Optional('states'): syaml.MapPattern(syaml.Str(), syaml.Str()),
+    syaml.Optional('units'): syaml.Map({
+        'key': syaml.Str(),
+        'values': syaml.MapPattern(syaml.Str(), syaml.Str())
+    }),
+    syaml.Optional('unit_values'): syaml.MapPattern(syaml.Str(), syaml.Str())
+    })
+
+_COMPLIANT = 'COMPLIANT'
+
+def _load_yaml_with_schema(filepath, schema):
+  """Loads an instance YAML file and parses
+  it based on a syaml-formatted YAML schema.
+
+  Args:
+    filepath: filepath location of the YAML file
+    schema: YAML schema in syaml format
+
+  Returns:
+    Returns the parsed YAML data in a stricyaml-provided
+    datastructure which is similar to a Python dictionary.
+  """
+  f = open(filepath)
+  content = f.read()
+  f.close()
+
+  try:
+    parsed = syaml.load(content, schema)
+
+    return parsed
+  except Exception:
+    return None
+
+def parse_yaml(filename):
+  """Loads an instance YAML file and parses it with
+  multiple strictyaml-formatted YAML schemas. Expected format:
+  github.com/google/digitalbuildings/blob/master/ontology/docs/building_config.md#config-format
+
+  Args:
+    filename: filepath location of the YAML file
+
+  Returns:
+    Returns the parsed YAML data in a stricyaml-provided
+    datastructure which is similar to a Python dictionary.
+  """
+  yaml = _load_yaml_with_schema(filename, _SCHEMA)
+
+  if yaml is None:
+    return None
+
+  top_name = yaml.keys()[0]
+
+  if 'translation' in yaml[top_name].keys():
```

Comment:
make `translation` a constant

Reply:
and move it up to the top of the file or create a constants.py

Reply:
Done